OMR compile throws error with gcc 4.9.3

Reproducible on ubuntu16, gcc 4.9.3.

In file included from ../../omr/port/unix/omrsignal.c:63:0:
../../omr/include_core/omrsig.h:110:54: error: 'struct sigvec' declared inside parameter list [-Werror]
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;

--- a/include_core/omrsig.h
+++ b/include_core/omrsig.h
@@ -105,9 +105,13 @@ sighandler_t bsd_signal(int signum, sighandler_t handler) __THROW;
 sighandler_t sysv_signal(int signum, sighandler_t handler) __THROW;
 #if !defined(AIXPPC)
 #if defined(OSX)
+#if defined(OMR_OMRSIG_HAS_SIGVEC)
 int sigvec(int sig, struct sigvec *, struct sigvec *);
+#endif /* OMR_OMRSIG_HAS_SIGVEC */
 #else /* defined(OSX) */
+#if defined(OMR_OMRSIG_HAS_SIGVEC)
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
+#endif /* OMR_OMRSIG_HAS_SIGVEC */
 #endif /* defined(OSX) */
 #endif /* !defined(AIXPPC) */
 #endif /* !defined(J9ZOS390) */
@@ -128,5 +132,7 @@ int __sigactionset(size_t newct, const __sigactionset_t newsets[], size_t *oldct
 #endif  /* __cplusplus */
 
 #if defined(AIXPPC)
+#if defined(OMR_OMRSIG_HAS_SIGVEC)
 int sigvec(int sig, const struct sigvec *vec, struct sigvec *ovec) __THROW;
+#endif /* OMR_OMRSIG_HAS_SIGVEC */
 #endif /* defined(AIXPPC) */

